internal/subroutine/authorization_model_generation_integration_test.go

```diff
@@ -0,0 +1,456 @@
+func (suite *ConnectionTestSuite) SetupSuite() {
+	suite.T().Logf("Setting up test suite...")
+
+	suite.scheme = runtime.NewScheme()
+	utilruntime.Must(accountv1alpha1.AddToScheme(suite.scheme))
+	utilruntime.Must(kcptenancyv1alpha1.AddToScheme(suite.scheme))
+	utilruntime.Must(kcpcorev1alpha1.AddToScheme(suite.scheme))
+	utilruntime.Must(kcpapiv1alpha1.AddToScheme(suite.scheme))
+	utilruntime.Must(apiextensionsv1.AddToScheme(suite.scheme))
+	utilruntime.Must(securityv1alpha1.AddToScheme(suite.scheme))
+
+	useExistingKcp := true
+	if envValue, err := strconv.ParseBool(os.Getenv("USE_EXISTING_KCP")); err == nil {
+		useExistingKcp = envValue
+	}
+	var restConfig *rest.Config
+
+	if useExistingKcp {
+		kubeconfigPath := os.Getenv("KCP_KUBECONFIG")
+		if kubeconfigPath == "" {
+			suite.T().Skipf("skipping integration test: KCP_KUBECONFIG environment variable is required when USE_EXISTING_KCP=true")
+			return
+		}
+
+		suite.T().Logf("Using existing KCP cluster (USE_EXISTING_KCP=true)")
+		suite.T().Logf("Loading kubeconfig from: %s", kubeconfigPath)
+
+		kubeconfig, err := clientcmd.LoadFromFile(kubeconfigPath)
+		if err != nil {
+			suite.T().Skipf("skipping integration test: unable to load KCP kubeconfig from %s: %v", kubeconfigPath, err)
+			return
+		}
+
+		clientConfig := clientcmd.NewDefaultClientConfig(*kubeconfig, nil)
+		originalConfig, err := clientConfig.ClientConfig()
+		if err != nil {
+			suite.T().Skipf("skipping integration test: unable to create client config from kubeconfig %s: %v", kubeconfigPath, err)
+			return
+		}
+
+		restConfig = rest.CopyConfig(originalConfig)
+
+		// The envtest package expects the base URL without /clusters/ path
+		if strings.Contains(restConfig.Host, "/clusters/") {
+			parsed, err := url.Parse(restConfig.Host)
+			if err != nil {
+				suite.T().Skipf("skipping integration test: unable to parse host URL %s: %v", restConfig.Host, err)
+				return
+			}
+
+			parsed.Path = ""
+			restConfig.Host = parsed.String()
+			suite.T().Logf("Stripped workspace path from host, using base URL: %s", restConfig.Host)
+		}
+	} else {
+		suite.T().Logf("Starting local KCP server (USE_EXISTING_KCP=false or unset)")
+	}
+
+	suite.testEnv = &envtest.Environment{
+		Scheme:          suite.scheme,
+		Config:          restConfig,
+		UseExistingKcp:  &useExistingKcp,
+		AttachKcpOutput: os.Getenv("TEST_ATTACH_KCP_OUTPUT") == "true",
+	}
+
+	var err error
+	suite.rootConfig, err = suite.testEnv.Start()
+	if err != nil {
+		suite.T().Skipf("skipping integration tests: unable to start KCP: %v", err)
+		return
+	}
+
+	suite.T().Logf("KCP test environment started successfully")
+}
+
+func (suite *ConnectionTestSuite) TearDownSuite() {
+	if suite.testEnv != nil {
+		if err := suite.testEnv.Stop(); err != nil {
+			suite.T().Logf("error stopping test environment: %v", err)
+		}
+	}
+}
+
+func (suite *ConnectionTestSuite) TestOrganizationCreation() {
+	ctx := context.Background()
```

Comment:
```suggestion
	ctx := suite.T().Context()
```

[nip] use a context bound to the test suite